ensime_shared/ensime.py

```diff
@@ -944,9 +812,7 @@ def teardown(self):
 
     def current_client(self, quiet, bootstrap_server, create_client):
         """Return the current client for a given project."""
-        # Use current_file command because we cannot access self.vim
-        current_file_cmd = commands["current_file"]
-        current_file = self.vim.eval(current_file_cmd)
+        current_file = self.vim.current.buffer.name
```

Comment:
I didn't understand the comment here because it still uses `self.vim` anyway 😩   This change seems to work fine, but let me know if anyone sees an issue.

The cases I believe we know of where using the `vim` object can be problematic:

- Off of the main thread. The Python Vim API isn't threadsafe so that's why the only code that tries to invoke it from the polling worker thread, which is `disable_plugin`, jumps through the `threadsafe_vim` hoop (which is just YOLO for standard Vim…).
- In `Ensime.__init__`. We observed that using `self.vim` there didn't work and my hunch was that's because the class is first being defined when an autocommand like `VimEnter` fires triggering it to be loaded, and we're also *defining the code that Vim is trying to invoke for our autocommand handler* within the `Ensime` class, so there's a reentrancy thing going on there.

Neither of these apply here. It may have been that at one time, further back than I have spelunked in `git blame`, that `current_client` was being called from `__init__`?